### ISSUE TITLE
Symlink disable causing an error

### ISSUE THREAD

**Issue by pursehouse** [composer/composer#7379]
I'm having this issue in Windows 10 Bash.
`Composer version 1.6.5 2018-05-04 11:44:59`

I already tried deleting the composer.lock and deleted the links that were there before I added the option.

running `composer install` this gives me an error:

```
Loading composer repositories with package information
[ErrorException]
stream_context_create(): options should have the form ["wrappername"]["optionname"] = $value
```

with the config...

```
"repositories": {
    "vendor/package": {
        "type": "path",
        "url": "/path/to/vendor/package",
        "options" : {
            "symlink" : false
        }
    }
}
```

#5048 didn't seem to have the solution

any ideas? Thanks!

**Comment by pursehouse** [comment]
problem solved! I had it in several repo's and accidentally added it to a VCS repo as well. Perhaps the composer error display could use an upgrade in that situation?

**Comment by Seldaek** [comment]
Yeah we pass-through options straight to the stream context for VCS repos. It's kinda hard to validate this as it has many possible options.

**Comment by pursehouse** [comment]
@Seldaek but it could at least output more info of what data was involved with the error, I guess just an upgrade of the generic error display system? I've not looked at the internals. But to at least show the strings/arrays involved at that moment would sure solve a lot of headaches

**Comment by Seldaek** [comment]
Sure I guess this could be done relatively easily in StreamContextFactory where I assume the error originated from.

**Comment by pursehouse** [comment]
wonderful! thanks

**Comment by typoworx-de** [comment]
I also had this issue after update to latest composer release (Composer version 1.9.1 2019-11-01 17:20:17). In my previous older release I also noticed this message as "warning". Since update is was a breaking error.

I noticed I had to remove `"options": { "symlink": true }` from remote-repositories. For development I also use a local-vendor repository at the moment, which doesn't seem to be affected (symlink: true does not throw error at the moment).

I think composer should throw a more informative error-message being able to note where the real problem is.

**Comment by Seldaek** [comment]
Cannot repro anymore, so closing. If you still see this please share a composer.json that reproduces it.

**Comment by PascalBrouwers** [comment]
adding the package with `--prefer-source` also seemed to be required
